pkg/controller/clusters/status.go

```diff
@@ -117,6 +118,19 @@ func updateControlPlaneReadyCondition(cluster *anywherev1.Cluster, kcp *controlp
 		return
 	}
 
+	// We check the condition signifying the overall health of the control plane components. Usually, the control plane should be healthy
+	// at this point but if that is not the case, we report it as an error.
+	kcpControlPlaneHealthyCondition := conditions.Get(kcp, controlplanev1.ControlPlaneComponentsHealthyCondition)
+	if kcpControlPlaneHealthyCondition == nil {
```

Comment:
I don't think this is necessary correct
Given we don't know what the logic around this is, if it's not present, I would continue as if it was true. It just seems safer.